### PR TITLE
gh-107453: Document errno.{ECANCELED,EOWNERDEAD,ENOTRECOVERABLE,ENOTSUP}

### PR DESCRIPTION
**Pull request by qqwqqw689**
----
:books: Documentation preview :books:: https://cpython-previews--107486.org.readthedocs.build/

* Issue: gh-107453

### CODE REVIEW

**Comment by qqwqqw689** [comment]
#107453

File: Doc/library/errno.rst [hunk 1]
```diff
@@ -666,3 +666,19 @@ defined by the module.  The specific list of defined symbols is available as
    .. availability:: WASI, FreeBSD
 
    .. versionadded:: 3.11.1
+
+.. data:: ECANCELED
+
+   operation canceled
```
Comment (AA-Turner) [inline Doc/library/errno.rst]:
```suggestion
   Operation canceled

   .. versionadded:: 3.2
```

File: Doc/library/errno.rst [hunk 2]
```diff
@@ -666,3 +666,19 @@ defined by the module.  The specific list of defined symbols is available as
...
+
+.. data:: EOWNERDEAD
+
+   owner dead
```
Comment (AA-Turner) [inline Doc/library/errno.rst]:
```suggestion
   Owner died

   .. versionadded:: 3.2
```

File: Doc/library/errno.rst [hunk 3]
```diff
@@ -666,3 +666,19 @@ defined by the module.  The specific list of defined symbols is available as
...
+
+.. data:: ENOTRECOVERABLE
+
+   state not recoverable
```
Comment (AA-Turner) [inline Doc/library/errno.rst]:
```suggestion
   State not recoverable

   .. versionadded:: 3.2
```

File: Doc/library/errno.rst [hunk 4]
```diff
@@ -666,3 +666,19 @@ defined by the module.  The specific list of defined symbols is available as
...
+
+.. data:: ENOTSUP
+
+   not supported
```
Comment (AA-Turner) [inline Doc/library/errno.rst]:
```suggestion
   Operation not supported

   .. versionadded:: 3.2

```

**Comment by qqwqqw689** [comment]
@AA-Turner I think sorting is dispensable because we can use "Ctrl+f" .😉

**Review by hugovk — CHANGES_REQUESTED** [review]
Please put the constants in order. Then people can either ctrl+F, or see things where they expect when they don't.

**Comment by AA-Turner** [comment]
@hugovk -- mea culpa! I should have looked at linux's ordering before leaving a comment, it transpires only one constant was out-of-order (``ENOTSUP``, which is seemingly duplicates ``EOPNOTSUPP`` ([see errno(3)](https://man7.org/linux/man-pages/man3/errno.3.html)). As it was trivial, I've pushed a commit to fix, and reverted another inadvertent commit on the branch.

A

**Comment by AA-Turner** [comment]
> But it seems that linux doesn't define ENOTSUP.

From the man page linked:

<dl>
<dt>EOPNOTSUPP</dt>
<dd>Operation not supported on socket (POSIX.1-2001).<br>
(ENOTSUP and EOPNOTSUPP have the same value on Linux, but
according to POSIX.1 these error values should be
distinct.)</dd>
</dl>

A

**Comment by qqwqqw689** [comment]
Yes, I found it from https://lists.gnu.org/archive/html/bug-glibc/2002-08/msg00017.html

        [ENOTSUP] "Not supported".  A function returns this error when
        certain parameter values are valid, but the functionality they request
        is not available. This can mean that the function does not implement
        a particular command or option value or flag bit at all.  For functions
        that operate on some object given in a parameter, such as a file
        descriptor or a port, it might instead mean that only that specific
        object (file descriptor, port, etc.) is unable to support the other
        parameters given; different file descriptors might support different
        ranges of parameter values.

        If the entire function is not available at all in the implementation,
        it returns ENOSYS instead.

        [EOPNOTSUPP] "Operation not supported". The operation you requested is
        not supported.  Some socket functions don't make sense for all types of
        sockets, and others may not be implemented for all communications
        protocols.  In the GNU system, this error can happen for many calls when
        the object does not support the particular operation; it is a generic
        indication that the server knows nothing to do for that call.

**Review by hugovk — APPROVED** [review]

**Pull request merged**